crates/bevy_xr/src/camera.rs

```diff
@@ -79,26 +79,31 @@ impl CameraProjection for XrProjection {
             / (self.projection_matrix.to_cols_array()[10] + 1.0)
     }
 
-    // TODO calculate this properly
-    fn get_frustum_corners(&self, _z_near: f32, _z_far: f32) -> [Vec3A; 8] {
-        let ndc_corners = [
-            Vec3A::new(1.0, -1.0, 1.0),   // Bottom-right far
-            Vec3A::new(1.0, 1.0, 1.0),    // Top-right far
-            Vec3A::new(-1.0, 1.0, 1.0),   // Top-left far
-            Vec3A::new(-1.0, -1.0, 1.0),  // Bottom-left far
-            Vec3A::new(1.0, -1.0, -1.0),  // Bottom-right near
-            Vec3A::new(1.0, 1.0, -1.0),   // Top-right near
-            Vec3A::new(-1.0, 1.0, -1.0),  // Top-left near
-            Vec3A::new(-1.0, -1.0, -1.0), // Bottom-left near
-        ];
-
-        let mut view_space_corners = [Vec3A::ZERO; 8];
-        let inverse_matrix = self.projection_matrix.inverse();
-        for (i, corner) in ndc_corners.into_iter().enumerate() {
-            view_space_corners[i] = inverse_matrix.transform_point3a(corner);
+    fn get_frustum_corners(&self, z_near: f32, z_far: f32) -> [Vec3A; 8] {
+        fn normalized_corner(inverse_matrix: &Mat4, near: f32, ndc_x: f32, ndc_y: f32) -> Vec3A {
+            let clip_pos = Vec4::new(ndc_x * near, ndc_y * near, near, near);
+            // I don't know why multiplying the Z axis by -1 is necessary.
+            // As far as I can tell from (likely my incorrect understanding of the code),
+            // PerspectiveProjection::get_frustum_corners() has the Z axis inverted??
+            Vec3A::from_vec4(inverse_matrix.mul_vec4(clip_pos)) / near * Vec3A::new(1., 1., -1.)
```

Comment:
Is this related to infinite far plane (`far_z <= near_z`)?

Reply:
No, the reverse Z only flips the resulting depth in the range 0-1. It should still have a right-handed coordinate system with Z- forward, but the existing bevy math writes the resulting frustum as if it's in the Z+ space.